modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/ActionsComponentPropsTransformer.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import BridgeComponent from './utils/BridgeComponent';
+import Sidebar from './components/Sidebar';
+
+export default function propsTransformer({
+	items,
+	sidebarContainerSelector,
+	...otherProps
+}) {
+	const actions = {
+		showInfo() {
+			showSidebar();
+		}
+	}
+
+	const hideSidebar = () => {
+		sidebarRef.setState({open: false})
+	};
+
+	const showSidebar = () => {
+		sidebarRef.setState({open: true})
+	};
+
+	const sidebarRef = ReactDOM.render(
+		<BridgeComponent 
+			bridgedComponent={Sidebar}
+			onClose={hideSidebar}
+		/>,
+		document.querySelector(sidebarContainerSelector)
```

Comment:
So what is this `<BridgeComponent>` for exactly? I don't get the purpose of the indirection. Sorry for the silly question.

Reply:
There's probably a better way to do what I wanted. The idea is to get the instance of the component so we can manipulate the sidebar (open/close), but if the component is a function component we get no instance, so this is kind of a converter...

Reply:
I just read your comment
> In any case, you should not rely on the return value of ReactDOM.render():

Reply:
It does feel a bit fishy to mount something into the DOM as a side-effect of transforming the props, as opposed to lazily creating the sidebar in response to an action being selected.

Reply:
gottcha

Reply:
About the BridgeComponent... I still have the same problem, if I directly render the Sidebar I don't have any instance to call save, right?

Reply:
You could perhaps use [`useImperativeHandle()`](https://reactjs.org/docs/hooks-reference.html#useimperativehandle) to give you an "instance"-like handle on a functional component like `<Sidebar/>`.

That would be explicit, at least, because you'd be declaring your "imperative" public API. The current version in this PR is violating encapsulation, because you're calling `setState` from outside the component.

Does this help?

![Screenshot 2020-07-15 -171002-LjnQ8e4f@2x](https://user-images.githubusercontent.com/7074/87562268-15337080-c6be-11ea-978b-8f9aa5044c7c.png)

Reply:
YES

Reply:
@carloslancha and @wincent sitting in a tree...